src/wallet/rpc/addresses.cpp

```diff
@@ -553,7 +637,11 @@ RPCHelpMan getaddressinfo()
     LOCK(pwallet->cs_wallet);
 
     std::string error_msg;
-    CTxDestination dest = DecodeDestination(request.params[0].get_str(), error_msg);
+
+    auto decoded = DecodeDestinationIndicatingSP(request.params[0].get_str(), error_msg);
+    CTxDestination dest{std::get<0>(decoded)};
+    bool silent_payment{std::get<1>(decoded)};
+    int32_t silent_payment_index{std::get<2>(decoded)};
```

Comment:
8a757e61b94f76f83bbcbe449edda710187af5f3
use structured binding

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/24897/commits/7eecc8732b267c421b45166ba5a0576e602892f0.